packages/java-packages/codesnippet-maven-plugin/src/main/java/com/azure/tools/codesnippetplugin/implementation/SnippetReplacer.java

```diff
@@ -138,41 +138,47 @@ private static void runCodesnippets(Path codesnippetRootDirectory, String codesn
             return;
         }
 
-        // Get the files that match the codesnippet glob and are contained in the codesnippet root directory.
-        List<Path> codesnippetFiles = globFiles(codesnippetRootDirectory, codesnippetGlob, false);
-
         // Only get the source files if sources are included in the update.
         List<Path> sourceFiles = Collections.emptyList();
-        if (includeSources) {
+        if (includeSources && sourcesRootDirectory.toFile().exists()) {
             // Get the files that match the sources glob and are contained in the sources root directory.
             sourceFiles = globFiles(sourcesRootDirectory, sourcesGlob, true);
         }
 
+        // Only get the README files if READMEs are included in the update.
+        List<Path> readmeFiles = Collections.emptyList();
+        if (includeReadme && readmeRootDirectory.toFile().exists()) {
+            readmeFiles = globFiles(readmeRootDirectory, readmeGlob, true);
+        }
+
+        if (sourceFiles.isEmpty() && readmeFiles.isEmpty()) {
+            logger.info("No files to update.");
+            return;
+        }
```

Comment:
Should we move this check to the top before all other checks?

Reply:
This is effectively the first check aside from the plugin being configured to skip running. Current logic is:

- Check if plugin should skip running
- Attempt to locate Java source files to update
- Attempt to locate README files to update
- Check if there are any files to update
- Run replacement

So, nothing will run before this